pages/apis/graphql/graphql_resource_limits.md

```diff
@@ -0,0 +1,174 @@
+# GraphQL resource limits
+
+To make sure that Buildkite stays stable for everyone, we have set some limits on our GraphQL API. These limits prevent excessive or abusive calls to our servers while still allowing API users to use GraphQL endpoints in a wide range of ways.
+
+Our limits are based on the complexity of the query. We calculate complexity based requested resources of the query and encourage developers to responsibly use techniques for limiting calls, pagination, caching and retrying requests to lower the complexity of the queries.
+
+
+## Query complexity
+Every field type in the schema has an integer cost assigned to it. The cost of the query is the sum of the cost of each field. Usually, running the query is the best way to know the true cost of the query.
+
+A cost is based on what the field returns.
+
+<table>
+  <tr>
+    <td>Scalar</td>
+    <td>0 complexity point</td>
+  </tr>
+  <tr>
+    <td>Enum</td>
+    <td>0 complexity point</td>
+  </tr>
+  <tr>
+    <td>Object</td>
+    <td>1 complexity point</td>
+  </tr>
+  <tr>
+    <td>Interface</td>
+    <td>1 complexity point</td>
+  </tr>
+  <tr>
+    <td>Union</td>
+    <td>1 complexity point</td>
+  </tr>
+</table>
+
+Although these default costs are in place, Buildkite also reserves the right to set manual costs to individual fields.
+
+
+## Complexity calculation
+
+Buildkite calculates the cost of the query before and after the query execution.
+
+### Requested complexity
+The requested complexity is based on the number of fields and objects requested. Usually, requesting a deeply nested query or excluding pagination details from connections results in high requested complexity.
+
+A simple query like the following would incur more than 500 requested complexity points as the query asks for 503 possible resources.
+
+```graphql
+query RecentPipelineSlugs {
+  organization(slug: "organization-slug") {  # 1 point
+    pipelines(first: 500) {                  # 1 point
+      edges {                                # 1 point
+        node {                               # 500 points
+          slug                               # 0 point
+        }
+      }
+    }
+  }
+}
+```
+
+### Actual complexity
+The actual complexity is based on the results returned after the query execution, since the connection fields can return fewer nodes than requested. Lowering requested complexity usually lowers the actual complexity of queries.
```

Comment:
```suggestion
The actual complexity is based on the results returned after the query execution, since the connection fields can return fewer nodes than requested. Lowering the requested complexity usually lowers the actual complexity of queries.
```